packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```
@@ -223,6 +323,14 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
         }
     }
 
+    private async createOrder() {
+        const { getFieldsValues } = this.paypalcommerce || {};
+
+        return this.paypalCommerceIntegrationService.createOrder('paypalcommercecheckout', {
+            shouldSaveInstrument: getFieldsValues?.().shouldSaveInstrument || false,
```

Same here regarding object destructuring.

---

It is not necessary here

---

What happens if developer provide `string` instead of `function` in `getFieldsValues`?

---

Should we throw an error if `getFieldsValues` data type is not correct?

---

Updated this part of the code